gen/utils/data/class.go

```diff
@@ -0,0 +1,175 @@
+package data
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"unicode"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+type Class struct {
+	// This is used to prevent the deletion of the class if it is not allowed on APIC.
+	AllowDelete bool
+	// Full name of the class, ex "fvTenant".
+	ClassName string
+	// Capitalized name of the class, ex "FvTenant".
+	ClassNameForFunctions string
+	// Package part of the class, ex "fv".
+	ClassNamePackage string
+	// Name part of the class, ex "Tenant".
+	ClassNameShort string
+	// List of all child classes which are included inside the resource.
+	// When looping over maps in golang the order of the returned elements is random, thus list is used for order consistency.
+	Children []Child
+	// List of all possible parent classes.
+	ContainedBy []string
+	// Documentation specific information for the class.
+	Documentation ClassDocumentation
+	// List of all identifying properties of the class.
+	// These are properties that are part if the relative name (RN) format, ex 'tn-{name}' where name is the identifying property.
+	IdentifiedBy []string
+	// Indicates that the class is migrated from previous version of the provider.
+	// This is used to determine if legacy attributes to be exposed in the resource.
+	// This is used to determine if the plan modification logic.
+	IsMigration bool
+	// Indicates that the class is a relationship class.
+	IsRelational bool
+	// Indicates that when the class is included in a resource as a child it can only be configured once.
+	// This is used to determine the type of the nested attribute to be a map or list.
+	IsSingleNested bool
+	// Indicates for which APIC platform the class is available, cloud vs onprem.
+	// TODO: investigate enum type for this.
+	Platform string
+	// A map containing all the properties of the class.
+	Properties map[string]*Property
+	// Each property is categorised in all, required, optional, or read-only.
+	// When looping over maps in golang the order of the returned elements is random, thus lists are used for order consistency.
+	PropertiesAll      []string
+	PropertiesRequired []string
+	PropertiesOptional []string
+	PropertiesReadOnly []string
+	// The raw data from the meta file.
+	// Storing the raw data proactively in case we need to access the data at a later stage.
+	RawMetaData map[string]interface{}
```

Comment:
Is Raw if it is a map[string]interface{}?

Reply:
what do you mean?